### PR TITLE
Ensure HIP targets are globally visible

### PR DESCRIPTION
**Pull request by mcfadden8**

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]

File: cmake/thirdparty/SetupHIP.cmake [hunk 1]
```diff
@@ -52,7 +52,8 @@ endif()
 
 blt_import_library(NAME       blt_hip
                    COMPILE_FLAGS "--rocm-path=${ROCM_PATH}"
-                   EXPORTABLE ${BLT_EXPORT_THIRDPARTY})
+                   EXPORTABLE ${BLT_EXPORT_THIRDPARTY}
+                   GLOBAL On)
```
Comment (white238) [inline cmake/thirdparty/SetupHIP.cmake:56]:
```suggestion
                   GLOBAL ON)
```

File: cmake/thirdparty/SetupHIP.cmake [hunk 2]
```diff
@@ -63,7 +64,8 @@ add_library(blt::hip ALIAS blt_hip)
 blt_import_library(NAME          blt_hip_runtime
                    INCLUDES ${HIP_INCLUDE_DIRS}
                    TREAT_INCLUDES_AS_SYSTEM ON
-                   EXPORTABLE    ${BLT_EXPORT_THIRDPARTY})
+                   EXPORTABLE    ${BLT_EXPORT_THIRDPARTY}
+                   GLOBAL On)
```
Comment (white238) [inline cmake/thirdparty/SetupHIP.cmake:68]:
```suggestion
                   GLOBAL ON)
```

**Comment by white238** [comment]
Minor stylistic nitpick aside, Thanks!

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
Any idea why this was needed to be global but the others (I checked `mpi`) do not?

**Comment by davidbeckingsale** [comment]
Cannot make an alias to a non-global imported target. I was so focused on making sure the TPL export worked I didn't test the reverse.

**Pull request merged**